test/functional/feature_backwards_compatibility.py

```diff
@@ -0,0 +1,335 @@
+        node_v19_wallets_dir = os.path.join(node_v19.datadir, "regtest/wallets")
+        node_v18_wallets_dir = os.path.join(node_v18.datadir, "regtest/wallets")
+        node_v17_wallets_dir = os.path.join(node_v17.datadir, "regtest/wallets")
+        node_master.unloadwallet("w1")
+        node_master.unloadwallet("w2")
+        node_v19.unloadwallet("w1_v19")
+        node_v19.unloadwallet("w2_v19")
+        node_v18.unloadwallet("w1_v18")
+        node_v18.unloadwallet("w2_v18")
+
+        # Copy wallets to v0.17
+        for wallet in os.listdir(node_master_wallets_dir):
+            shutil.copytree(
+                os.path.join(node_master_wallets_dir, wallet),
+                os.path.join(node_v17_wallets_dir, wallet)
+            )
+        for wallet in os.listdir(node_v18_wallets_dir):
+            shutil.copytree(
+                os.path.join(node_v18_wallets_dir, wallet),
+                os.path.join(node_v17_wallets_dir, wallet)
+            )
+
+        # Copy wallets to v0.18
+        for wallet in os.listdir(node_master_wallets_dir):
+            shutil.copytree(
+                os.path.join(node_master_wallets_dir, wallet),
+                os.path.join(node_v18_wallets_dir, wallet)
+            )
+
+        # Copy wallets to v0.19
+        for wallet in os.listdir(node_master_wallets_dir):
+            shutil.copytree(
+                os.path.join(node_master_wallets_dir, wallet),
+                os.path.join(node_v19_wallets_dir, wallet)
+            )
+
+        # Open the wallets in v0.19
+        node_v19.loadwallet("w1")
+        wallet = node_v19.get_wallet_rpc("w1")
+        info = wallet.getwalletinfo()
+        assert(info['private_keys_enabled'])
+        assert(info['keypoolsize'] > 0)
+        txs = wallet.listtransactions()
+        assert_equal(len(txs), 5)
+        assert_equal(txs[1]["txid"], tx1_id)
+        assert_equal(txs[2]["walletconflicts"], [tx1_id])
+        assert_equal(txs[1]["replaced_by_txid"], tx2_id)
+        assert(not(txs[1]["abandoned"]))
+        assert_equal(txs[1]["confirmations"], -1)
+        assert_equal(txs[2]["blockindex"], 1)
+        assert(txs[3]["abandoned"])
+        assert_equal(txs[4]["walletconflicts"], [tx3_id])
+        assert_equal(txs[3]["replaced_by_txid"], tx4_id)
+        assert(not(hasattr(txs[3], "blockindex")))
+
+        node_v19.loadwallet("w2")
+        wallet = node_v19.get_wallet_rpc("w2")
+        info = wallet.getwalletinfo()
+        assert(info['private_keys_enabled'] == False)
+        assert(info['keypoolsize'] == 0)
+
+        node_v19.loadwallet("w3")
+        wallet = node_v19.get_wallet_rpc("w3")
+        info = wallet.getwalletinfo()
+        assert(info['private_keys_enabled'])
+        assert(info['keypoolsize'] == 0)
+
+        # Open the wallets in v0.18
+        node_v18.loadwallet("w1")
+        wallet = node_v18.get_wallet_rpc("w1")
+        info = wallet.getwalletinfo()
+        assert(info['private_keys_enabled'])
```

Comment:
style-nit: 
```suggestion
        assert info['private_keys_enabled']
```

(Applies throughout this file)

Reply:
Done